src/main/java/appland/telemetry/data/BaseData.java

```diff
@@ -0,0 +1,42 @@
+package appland.telemetry.data;
+
+import java.util.HashMap;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+
+public class BaseData {
+    @SerializedName("ver")
+    @NotNull final Integer version = 2;
+
+    @SerializedName("name")
+    @NotNull String name;
+
+    @SerializedName("properties")
+    @Nullable HashMap<String, String> properties;
```

Comment:
nit: It's better to use an interface for the properties instead, `Map<String,Map>`. This makes it more generic and simplifies later changes.